ionc/inc/ion_debug.h

```diff
@@ -79,7 +79,7 @@ extern "C" {
  * If error has already happened before, the error code of the current function will be ignored.
  * If there're no previous errors, the return code of the current function will be kept.
  */
-#define UPDATEERROR(x)       { iERR errBackup = err; err = x; if (err == IERR_OK) { err = errBackup; }}
+#define UPDATEERROR(x)       { iERR errBackup = (x); if (err == IERR_OK) { err = errBackup; }}
```

Comment:
Unfortunately not, because `x` may be a function call which needs to be executed unconditionally, but its return value only stored if there isn't an existing error.

Reply:
Oh, I see.